File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cs
public class Solution {
    public int SubarraySum(int[] nums, int k) {
        var mp = new Dictionary<int, int>();
        
        
        int sum = 0, cnt = 0;
        mp[sum] = 1;
        
        for(int i = 0; i < nums.Length; i++) {
            sum += nums[i];
            
            
            if(mp.ContainsKey(sum - k)) 
                cnt += mp[sum - k];
            
        
            if(mp.ContainsKey(sum)){
                mp[sum]++;
            }else {
                mp[sum] = 1;
            }
        }
        
        return cnt;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        map<int, int> mp;
        mp[0]++;
        int n = size(nums), cnt = 0, sum = 0;
        
        for(int i = 0; i < n; i++) {
            sum += nums[i];
            cnt += mp[sum - k];
            mp[sum]++;
        }
        
        return cnt;
    }
};